web-app/HeimScripts/heatmap/run.R

```diff
@@ -104,6 +106,30 @@ buildFields <- function(df){
   return(df)
 }
 
+writeDataForZip <- function(df, patientIDs) {
+  pidCols    <- as.character(patientIDs)
+  t          <- df
+  t[pidCols] <- lapply(t[pidCols], function(v) { (v - df$MEAN) / df$SD })
+  allCols    <- c(c("Row.Label", "Bio.marker", "MEAN", "SD", "SIGNIFICANCE"), pidCols)
+  write.table(t[allCols], "heatmap_data.tsv", sep="\t", na="", row.names=F, col.names=T)
```

Comment:
are there any columns of data.frame 't' not used for writing to file? Otherwise 'write.table(t, "heatmap... suffices and the allColls variable can be omitted